packages/astro/src/cli/preferences/index.ts

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-console */
+import type yargs from 'yargs-parser';
+import type { AstroSettings } from '../../@types/astro.js';
+
+import { cyan } from 'kleur/colors';
+import { fileURLToPath } from 'node:url';
+
+import * as msg from '../../core/messages.js';
+import { createLoggerFromFlags, flagsToAstroInlineConfig } from '../flags.js';
+import { resolveConfig } from '../../core/config/config.js';
+import { createSettings } from '../../core/config/settings.js';
+import { isValidKey, type PreferenceKey } from '../../preferences/index.js';
+import { DEFAULT_PREFERENCES } from '../../preferences/defaults.js';
+import dlv from 'dlv';
+
+
+interface PreferencesOptions {
+	flags: yargs.Arguments;
+}
+
+const PREFERENCES_SUBCOMMANDS = ['get', 'set', 'enable', 'disable', 'delete', 'reset', 'list'] as const;
+export type Subcommand = typeof PREFERENCES_SUBCOMMANDS[number];
+
+function isValidSubcommand(subcommand: string): subcommand is Subcommand {
+	return PREFERENCES_SUBCOMMANDS.includes(subcommand as Subcommand);
+}
+
+export async function preferences(subcommand: string, key: string, value: string | undefined, { flags }: PreferencesOptions): Promise<number> {
+	if (!isValidSubcommand(subcommand) || flags?.help || flags?.h) {
+		msg.printHelp({
+			commandName: 'astro preferences',
+			usage: 'set [key] [:value]',
+			tables: {
+				Flags: [
+					['--global', 'Change setting value globally.'],
+					['--help (-h)', 'See all available flags.'],
+				],
+			},
```

Comment:
When I run `pnpm astro preferences --help`, it only shows the flags but not the additional subcommands. Should we also list them here?

Reply:
Yes we should! Thanks for catching that.